### DISCUSSION TITLE
FusedConv Error when using GPU

### DISCUSSION THREAD

**Discussion by zengqi0730** [snakers4/silero-vad#302] [Q&A] [answered]
When i run the vad model with onnxruntime using CUDAExecutionProvider, the error occurs:

onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : FAIL : Non-zero status code returned while running If node. Name:'If_25' Status Message: Non-zero status code returned while running FusedConv node. Name:'Conv_139' Status Message: CUDNN failure 3: CUDNN_STATUS_BAD_PARAM ; GPU=0 ; hostname=zq ; expr=cudnnAddTensor(Base::CudnnHandle(), &alpha, Base::s_.z_tensor, Base::s_.z_data, &alpha, Base::s_.y_tensor, Base::s_.y_data); 

It seems that the cuda ops can not run the fused node onnxruntime optimazed.

When i set  graph_optimization_level = GraphOptimizationLevel.ORT_ENABLE_BASIC , the inference works well. But the inference time is 4-5 times slower than CPUExecutionProvider. (averange 800us/frame for cuda: 150us/frame for cpu)

What's your suggestion about speeding up inference time using GPU devices?  At least the same as CPU performance！

**Comment by snakers4** [comment, marked as answer]
Hi, the VAD was not designed to run on GPU.
